### PR TITLE
Add support for WebAssembly Interface Types

### PR DESCRIPTION
**Pull request by eduardomourar**
## Description

Add support for WebAssembly Interface Type (WIT):
https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      -  https://github.com/search?q=path%3A*.wit&type=code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/WebAssembly/wasi-io/tree/e6bcf18ac5e2d2deee5016fc4b6e260fc14d4bdc/wit
    - Sample license(s): Apache 2.0
  - [x] I have included a syntax highlighting grammar: https://github.com/bytecodealliance/vscode-wit
  - [x] I have added a color
    - Hex value: `#6250e7`
    - Rationale: Based on the WebAssembly logo

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -7409,6 +7409,18 @@ Windows Registry Entries:
...
+  type: programming
+  color: "#6250e7"
+  extensions:
+  - ".wit"
+  aliases:
+  - wit
+  ace_mode: text
+  tm_scope: source.wit
+  codemirror_mode: clike
+  codemirror_mime_type: text/x-csrc
```
Comment (Alhadis) [inline lib/linguist/languages.yml:7422]:
```suggestion
  codemirror_mode: webidl
  codemirror_mime_type: text/x-webidl
```

CodeMirror's [C-like mode](https://codemirror.net/5/mode/clike/) fails to highlight the following WIT fragment:

~~~
world my-world {
	import host: self.host
	import other-functionality: pkg.sibling-file
	import more-functionality: pkg.sibling-file.specific-interface
}

interface host {
	// ...
}
~~~

Its [WebIDL mode](https://codemirror.net/5/mode/webidl/index.html), OTOH, does a better—albeit still imperfect—job of highlighting WIT source. Having some highlighting is better than none at all.

**Review by Alhadis — CHANGES_REQUESTED** [review]
This is the first grammar I've encountered that uses `repository` fields in contexts that aren't the top-level grammar object. I see that VSCode supports this, but Atom doesn't—and by extension, that (probably) includes PrettyLights (GitHub's implementation of TextMate's highlighting system, which is closed-source).

So comments will end up looking something like this on GitHub:

Figure 1

This can be solved by moving every named rule to the top-level `repository` object.

**Comment by eduardomourar** [comment]
The repository fields have been moved to the top-level object.

**Review by Alhadis — APPROVED** [review]
Alright, grammar's looking good. The syntax highlighting grammar, I mean, not that of my Dunglish:

Figure 1

I noticed a few omissions in your grammar, but nothing urgent. I'll submit an issue to `vscode-wit` later with more info.

**Review by lildude — APPROVED** [review]

**Pull request merged**